internal/provider/nordvpn/api/getservers.go

```diff
@@ -0,0 +1,80 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/qdm12/gluetun/internal/constants/vpn"
+	"github.com/qdm12/gluetun/internal/models"
+)
+
+var (
+	ErrHTTPStatusCodeNotOK = errors.New("HTTP status code not OK")
+)
+
+func (api *API) GetServers(recommended bool) (servers []models.Server, err error) {
```

Comment:
Pass in a context to this function so it can be injected to `NewRequestWithContext`

Something like

```suggestion
func (api *API) GetServers(ctx context.Context, recommended bool) (servers []models.Server, err error) {
```

Reply:
i had the idea of use this call for fetching the server and also selecting the server so we can use the server recommendation base on current server load, i believe one of those calls don't have a context and that is the reason that i just don't include it in general

Reply:
> i believe one of those calls don't have a context and that is the reason that i just don't include it in general

You should pretty much always use a context so any IO operation can be canceled promptly by the calling code.
In this case `NewRequestWithContext` takes a context (and `context.Background()` doesn't do anything).

> and also selecting the server so we can use the server recommendation base on current server load

that's going to be a tough one since gluetun is tightly coded to not allow any traffic before the vpn is up. I'm working on making this more configurable/better code though, but it will take some time.